plugins/openstack/src/test/java/org/eclipse/xpanse/plugins/openstack/OpenstackOrchestratorPluginTest.java

```diff
@@ -1,91 +1,434 @@
+import org.eclipse.xpanse.modules.models.service.deploy.enums.DeployResourceKind;
+import org.eclipse.xpanse.modules.orchestrator.manage.ServiceManagerRequest;
+import org.eclipse.xpanse.modules.orchestrator.monitor.ResourceMetricsRequest;
+import org.eclipse.xpanse.modules.orchestrator.monitor.ServiceMetricsRequest;
+import org.eclipse.xpanse.plugins.openstack.common.constants.OpenstackEnvironmentConstants;
+import org.eclipse.xpanse.plugins.openstack.manage.ServersManager;
+import org.eclipse.xpanse.plugins.openstack.monitor.MetricsManager;
+import org.eclipse.xpanse.plugins.openstack.resourcehandler.OpenstackTerraformResourceHandler;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
 
+@ExtendWith(MockitoExtension.class)
 class OpenstackOrchestratorPluginTest {
+    private final String terraformOpenStackVersion = "1.52.1";
+    @Mock
+    private OpenstackTerraformResourceHandler mockOpenstackTerraformResourceHandler;
+    @Mock
+    private MetricsManager mockMetricsManager;
+    @Mock
+    private ServersManager mockServersManager;
+
+    @InjectMocks
+    private OpenstackOrchestratorPlugin openstackOrchestratorPluginUnderTest;
 
-    private final OpenstackOrchestratorPlugin plugin = new OpenstackOrchestratorPlugin(
-            new OpenstackTerraformResourceHandler(), new MetricsManager(new KeystoneManager(null), new ResourcesService(),
-                    new GnocchiToXpanseModelConverter(),
-                    new AggregationService(new MetricsQueryBuilder()),
-                    new MeasuresService(new MetricsQueryBuilder()), null,
-                    new ServiceMetricsStore(new ServiceMetricsCacheManager())));
+    @BeforeEach
+    void setUp() {
+        ReflectionTestUtils.setField(openstackOrchestratorPluginUnderTest,
+                "terraformOpenStackVersion", terraformOpenStackVersion);
+    }
 
     @Test
-    void getResourceHandler() {
-        assertTrue(plugin.getResourceHandler() instanceof OpenstackTerraformResourceHandler);
+    void testGetResourceHandler() {
+        assertThat(openstackOrchestratorPluginUnderTest.getResourceHandler())
+                .isEqualTo(mockOpenstackTerraformResourceHandler);
     }
 
     @Test
-    void getCsp() {
-        assertEquals(Csp.OPENSTACK, plugin.getCsp());
+    void testGetCsp() {
+        assertThat(openstackOrchestratorPluginUnderTest.getCsp()).isEqualTo(Csp.OPENSTACK);
     }
 
     @Test
-    void getAvailableCredentialTypes() {
-        assertEquals(List.of(CredentialType.VARIABLES), plugin.getAvailableCredentialTypes());
+    void testRequiredProperties() {
+        assertThat(openstackOrchestratorPluginUnderTest.requiredProperties())
+                .isEqualTo(List.of(OpenstackEnvironmentConstants.AUTH_URL));
```

Comment:
@baixinsui this must be an empty list. This was changed with another PR which was merged today morning.

Reply:
Updated.